### PR TITLE
[TK-06834] Fix conductor is stuck at 31 read handles and deadlocks

### PR DESCRIPTION
**Pull request by freesig**
If more the 31 read handles are taken on a tokio `RwLock` then it won't allow any more to be taken. 
If we have a situation where > 31 reads are dependent then we have a deadlock.

Maybe this is unlikely but I was able to produce dependent reads by simply calling `get_links()` where a base is missing then the call gets routed to ourself because we are an authority and simply do this > 31 times concurrently and we have a deadlock.

Maybe it's not a super likely scenario but the problem is it's vague and could show up in other weird ways.

This PR now fixes the issue by not holding read locks for network events or zome calls

### CODE REVIEW

File: crates/holochain/src/conductor/p2p_store.rs [hunk 1]
```diff
@@ -207,6 +211,106 @@ pub fn exchange_peer_info(envs: Vec<EnvironmentWrite>) {
     }
 }
 
+/// Get agent info for a single agent
+pub fn get_agent_info_signed(
+    environ: EnvironmentWrite,
```
Comment (maackle) [inline crates/holochain/src/conductor/p2p_store.rs:216]: This pattern of just using a raw environment to get stuff out of the database is unprecedented, but it's fine especially if we're doing database refactor soon

File: crates/test_utils/wasm/wasm_workspace/multiple_calls/src/lib.rs [hunk 2]
```diff
@@ -37,3 +37,15 @@ fn get_entry_multiple(n: TestInt) -> ExternResult<TestBytes> {
 
     Ok(TestBytes(bytes))
 }
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, SerializedBytes)]
+pub struct TwoInt(pub u32, pub u32);
```
Comment (maackle) [inline crates/test_utils/wasm/wasm_workspace/multiple_calls/src/lib.rs:42]: This probably isn't being used

```suggestion
```

**Review by maackle — APPROVED** [review]
Nice, this feels like the right approach

File: crates/test_utils/wasm/wasm_workspace/multiple_calls/src/lib.rs (hunk 2 above)
Comment (freesig) [reply crates/test_utils/wasm/wasm_workspace/multiple_calls/src/lib.rs:42]: Why? It's used in the function below though

**Pull request merged**